unified-runtime/source/adapters/opencl/device.cpp

```diff
@@ -1418,6 +1418,59 @@ UR_APIEXPORT ur_result_t UR_APICALL urDeviceGetInfo(ur_device_handle_t hDevice,
     return ReturnValue(true);
   case UR_DEVICE_INFO_KERNEL_LAUNCH_CAPABILITIES:
     return ReturnValue(0);
+  case UR_DEVICE_INFO_LUID: {
+    // LUID is only available on Windows.
+    // Intel extension for device LUID. This returns the LUID as
+    // std::array<std::byte, 8>. For details about this extension,
+    // see sycl/doc/extensions/supported/sycl_ext_intel_device_info.md.
+
+    // Use the cl_khr_device_uuid extension, if available.
+    bool isKhrDeviceLuidSupported = false;
+    if (hDevice->checkDeviceExtensions({"cl_khr_device_uuid"},
+                                       isKhrDeviceLuidSupported) !=
+            UR_RESULT_SUCCESS ||
+        !isKhrDeviceLuidSupported) {
+      return UR_RESULT_ERROR_UNSUPPORTED_ENUMERATION;
+    }
+
+    cl_bool isLuidValid;
+    CL_RETURN_ON_FAILURE(
+        clGetDeviceInfo(hDevice->CLDevice, CL_DEVICE_LUID_VALID_KHR,
+                        sizeof(cl_bool), &isLuidValid, nullptr));
+
+    if (!isLuidValid) {
+      return UR_RESULT_ERROR_UNSUPPORTED_FEATURE;
```

Comment:
I think this should be UNSUPPORTED_ENUMERATION.

Reply:
It is quite confusing to me as to when UR_RESULT_ERROR_UNSUPPORTED_ENUMERATION vs UR_RESULT_ERROR_UNSUPPORTED_FEATURE should be used. 

I used `UR_RESULT_ERROR_UNSUPPORTED_FEATURE` in the other device implementations. The `UR_RESULT_ERROR_UNSUPPORTED_ENUMERATION` above this is used here as I took the check from UUID.

So if I do update this I should also update other uses of unsupported feature I added to unsupported enumeration.

Do you know the exact difference between the two?